pkg/operation/cloudbotanist/alicloudbotanist/controlplane.go

```diff
@@ -0,0 +1,177 @@
+	if err != nil {
+		return "", err
+	}
+	stateVariables, err := tf.GetStateOutputVariables(vpcID, vswitchID)
+	if err != nil {
+		return "", err
+	}
+
+	key := base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeyID])
+	secret := base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeySecret])
+	cfg := &cloudConfig{}
+	cfg.Global.KubernetesClusterTag = b.Shoot.SeedNamespace
+	cfg.Global.VpcID = stateVariables[vpcID]
+	cfg.Global.ZoneID = b.Shoot.Info.Spec.Cloud.Alicloud.Zones[0]
+	cfg.Global.VswitchID = stateVariables[vswitchID]
+	cfg.Global.AccessKeyID = key
+	cfg.Global.AccessKeySecret = secret
+
+	cfgJSON, err := json.Marshal(cfg)
+	if err != nil {
+		return "", err
+	}
+
+	return string(cfgJSON), nil
+}
+
+// RefreshCloudProviderConfig refreshes the cloud provider credentials in the existing cloud
+// provider config.
+// Not needed on Alicloud.
+func (b *AlicloudBotanist) RefreshCloudProviderConfig(currentConfig map[string]string) map[string]string {
+	existing := currentConfig[common.CloudProviderConfigMapKey]
+	cfg := &cloudConfig{}
+	if err := json.Unmarshal([]byte(existing), cfg); err != nil {
+		return currentConfig
+	}
+
+	cfg.Global.AccessKeyID = base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeyID])
+	cfg.Global.AccessKeySecret = base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeySecret])
+	newProviderCfg, err := json.Marshal(cfg)
+	if err == nil {
+		currentConfig[common.CloudProviderConfigMapKey] = string(newProviderCfg)
+	}
+
+	return currentConfig
+}
+
+// GenerateKubeAPIServerConfig generates the cloud provider specific values which are required to render the
+// Deployment manifest of the kube-apiserver properly.
+func (b *AlicloudBotanist) GenerateKubeAPIServerConfig() (map[string]interface{}, error) {
+	return nil, nil
+}
+
+// GenerateCloudControllerManagerConfig generates the cloud provider specific values which are required to
+// render the Deployment manifest of the cloud-controller-manager properly.
+func (b *AlicloudBotanist) GenerateCloudControllerManagerConfig() (map[string]interface{}, string, error) {
+	chartName := "alicloud-cloud-controller-manager"
+	conf := map[string]interface{}{
+		"configureRoutes": false,
+	}
+	newConf, err := b.InjectImages(conf, b.SeedVersion(), b.ShootVersion(), common.AlicloudControllerManagerImageName)
+	if err != nil {
+		return conf, chartName, err
+	}
+
+	return newConf, chartName, nil
+}
+
+// GenerateCSIConfig generates the configuration for CSI charts
+func (b *AlicloudBotanist) GenerateCSIConfig() (map[string]interface{}, error) {
+	conf := map[string]interface{}{
+		"credential": map[string]interface{}{
+			"accessKeyID":     base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeyID]),
+			"accessKeySecret": base64.StdEncoding.EncodeToString(b.Shoot.Secret.Data[AccessKeySecret]),
+		},
+		"enabled": true,
+	}
+	newConf, err := b.InjectImages(conf, b.SeedVersion(), b.ShootVersion(),
```

Comment:
```suggestion
	return b.InjectImages(conf, b.SeedVersion(), b.ShootVersion(), ...
```

instead of checking for error - unneeded